backend/inspirehep/search/api.py

```diff
@@ -157,37 +159,67 @@ def match_reference(self, query_string):
         if not query_string:
             return None
 
+        reference = self.query_string_to_reference_object_or_none(query_string)
+        if not reference:
+            return None
+
+        reference = self.normalize_reference_journal_title(reference)
+        reference = self.convert_old_publication_info_to_new(reference)
+
+        reference_match_control_number = match_reference_control_number(reference)
+        if reference_match_control_number:
+            return (
+                InspireSearch()
+                .params(version=True)
+                .query("term", control_number=reference_match_control_number)
+                .execute()
+            )
+        return None
+
+    def query_string_to_reference_object_or_none(self, query_string):
         try:
             reference = get_reference_from_grobid(query_string)
         except RequestException:
-            LOGGER.exception("GROBID request error", query_string=query_string)
-            return None
+            LOGGER.exception("Request error from GROBID.", query_string=query_string)
         except Exception:
             LOGGER.exception(
                 "Error processing reference from GROBID", query_string=query_string
             )
-            return None
-
-        if not reference:
-            return None
+        else:
+            return reference
+        return None
 
+    def normalize_reference_journal_title(self, reference):
         journal_title = get_value(reference, "reference.publication_info.journal_title")
         try:
             reference["reference"]["publication_info"][
                 "journal_title"
             ] = JournalsSearch().normalize_title(journal_title)
         except KeyError:
             pass
+        return reference
 
-        reference_match_control_number = match_reference_control_number(reference)
-        if reference_match_control_number:
-            return (
-                InspireSearch()
-                .params(version=True)
-                .query("term", control_number=reference_match_control_number)
-                .execute()
+    def convert_old_publication_info_to_new(self, reference):
+        try:
+            publication_info = force_list(
+                get_value(reference, "reference.publication_info", default=[])
             )
```

Comment:
```suggestion
            publication_info = [
                get_value(reference, "reference.publication_info", default={})
            ]
```
as the schema guarantees that this is never a list but a dict (`force_list` is useful if you can have either a list or a scalar and you want to make sure you get a list).